modules/sdk-core/src/bitgo/utils/tss/ecdsa/ecdsaMPCv2.ts

```diff
@@ -1168,4 +1179,140 @@ export class EcdsaMPCv2Utils extends BaseEcdsaUtils {
+          ovc: {
+            1: { bitgoToOvcMsg2: this.formatP2PMessage(result.bitgoToUserMsg2) },
+            2: { bitgoToOvcMsg2: this.formatP2PMessage(result.bitgoToBackupMsg2) },
+          },
+        },
+      },
+    };
+  }
+
+  public async keyGenSMCRound2(
+    enterprise: string,
+    payload: OVC2ToBitgoRound2Payload
+  ): Promise<BitgoToOVC1Round2Response> {
+    assert(
+      payload.state === KeyCreationMPCv2States.WaitingForBitgoRound2Data,
+      'Invalid state for round 2, got: ' + payload.state
+    );
+    const ovc1 = payload.ovc[1];
+    const ovc2 = payload.ovc[2];
+    const sessionId = payload.platform.sessionId;
+    const messages = { p2pMessages: [ovc1.ovcToBitgoMsg2, ovc2.ovcToBitgoMsg2], broadcastMessages: [] };
+    const result = await this.sendKeyGenerationRound2(enterprise, sessionId, messages);
+
+    return {
+      wallet: {
+        tssVersion: payload.tssVersion,
+        walletType: payload.walletType,
+        coin: payload.coin,
+        ovc: payload.ovc,
+        state: KeyCreationMPCv2States.WaitingForOVC1Round3aData,
+        platform: {
+          ...payload.platform,
+          sessionId: result.sessionId,
+          bitgoCommitment2: result.bitgoCommitment2,
+          ovc: {
+            1: { ...payload.platform.ovc[1], bitgoToOvcMsg3: this.formatP2PMessage(result.bitgoToUserMsg3) },
+            2: { ...payload.platform.ovc[2], bitgoToOvcMsg3: this.formatP2PMessage(result.bitgoToBackupMsg3) },
+          },
+        },
+      },
+    };
+  }
+
+  public async keyGenSMCRound3(
+    enterprise: string,
+    payload: OVC1ToBitgoRound3Payload
+  ): Promise<BitgoToOVC1Round3Response> {
+    assert(
+      payload.state === KeyCreationMPCv2States.WaitingForBitgoRound3Data,
+      'Invalid state for round 3, got: ' + payload.state
+    );
+    const ovc1 = payload.ovc[1];
+    const ovc2 = payload.ovc[2];
+    const sessionId = payload.platform.sessionId;
+    const messages = {
+      p2pMessages: [ovc1.ovcToBitgoMsg3, ovc2.ovcToBitgoMsg3],
+      broadcastMessages: [ovc1.ovcMsg4, ovc2.ovcMsg4],
+    };
+    const result = await this.sendKeyGenerationRound3(enterprise, sessionId, messages);
+
+    const keychains = this.baseCoin.keychains();
+    const bitgoKeychain = await keychains.add({
+      source: 'bitgo',
+      keyType: 'tss',
+      commonKeychain: result.commonKeychain,
+      isMPCv2: true,
+    });
+
+    return {
+      bitGoKeyId: bitgoKeychain.id,
+      wallet: {
+        tssVersion: payload.tssVersion,
+        walletType: payload.walletType,
+        coin: payload.coin,
+        ovc: payload.ovc,
+        state: KeyCreationMPCv2States.WaitingForOVC1GenerateKey,
+        platform: {
+          ...payload.platform,
+          commonKeychain: result.commonKeychain,
+          bitgoMsg4: this.formatBitgoBroadcastMessage(result.bitgoMsg4),
+        },
+      },
+    };
+  }
+
+  public async keyGenSMCAddOthersKeys(
```

Comment:
```suggestion
  public async keyGenOVCAddUserAndBackupKeys(
```

Reply:
I think the naming can be updated as well, `OtherKeys` is a little vague. (i.e. if I where reading the call site of this function, I have to read the function body to know what it's doing).

Reply:
Also how is this method intended to be used?

Reply:
This is not really necessary but an easy way for the clients to create the user and backup keys docs once the OVCs generation is done, then they can be used to generate a wallet.